services/tasks.js

```diff
@@ -26,6 +29,50 @@ const updateTaskStatusToDone = async (tasksData) => {
     return { ...summary };
   }
 };
+
+const addTaskCreatedAtAndUpdatedAtFields = async () => {
+  const operationStats = {
+    failedTasksIds: [],
+    totalFailedTasks: 0,
+    totalTasks: 0,
+  };
+  const updatedTasks = [];
+  const tasks = await tasksModel.get();
+
+  if (tasks.empty) {
+    return operationStats;
+  }
+
+  tasks.forEach(async (task) => {
+    const taskData = task.data();
+    taskData.createdAt = task.createTime.seconds;
+    taskData.updatedAt = task.updateTime.seconds;
+    updatedTasks.push({ id: task.id, data: taskData });
+  });
+
+  const multipleTasksUpdateBatch = [];
+  const chunkedTasks = chunks(updatedTasks, FIRESTORE_BATCH_OPERATIONS_LIMIT);
+
+  chunkedTasks.forEach(async (tasks) => {
+    const batch = firestore.batch();
+    tasks.forEach(({ id, data }) => {
+      batch.update(tasksModel.doc(id), data);
+    });
+    operationStats.totalTasks += tasks.length;
+    try {
+      await batch.commit();
+      multipleTasksUpdateBatch.push(batch);
```

Comment:
batch is promise here??

Reply:
Yes, the batch is a promise https://firebase.google.com/docs/firestore/manage-data/transactions#batched-writes